worker/src/main/java/org/apache/celeborn/service/deploy/worker/storage/hdfs/StreamsManager.java

```diff
@@ -0,0 +1,121 @@
+import com.google.common.cache.RemovalListener;
+import com.google.common.cache.RemovalNotification;
+import org.apache.celeborn.common.CelebornConf;
+import org.apache.celeborn.common.metrics.source.AbstractSource;
+import org.apache.hadoop.fs.FSDataOutputStream;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import java.io.IOException;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+
+public final class StreamsManager {
+    public static final Logger log = LoggerFactory.getLogger(StreamsManager.class);
+
+    LoadingCache<Path, FSDataOutputStream> cache;
+    private final CelebornConf conf;
+    private final FileSystem fileSystem;
+    private final int maxCapacity;
+    private final long maxStreamIdleMs;
+    private long totalGetkeyCount = 0;
+    private long totalLoaderCount = 0;
+    private final long maxCount = 1000000000000L;
+
+    private final int concurrentLevel;
+    public StreamsManager(CelebornConf conf, FileSystem fileSystem) {
+        this.conf = conf;
+        this.fileSystem = fileSystem;
+        this.maxCapacity = conf.workerOpenHDFSOutputStreamMax();
+        this.maxStreamIdleMs = conf.workerHDFSOutputStreamIdleMsMax();
+        this.concurrentLevel = conf.workerHDFSOutputStreamConcurrentLevel();
+        RemovalListener<Path, FSDataOutputStream> listener = new RemovalListener<Path, FSDataOutputStream>() {
+            @Override
+            public void onRemoval(RemovalNotification<Path, FSDataOutputStream> notification) {
+                if (notification.getValue() != null) {
+                    try {
+                        notification.getValue().close();
+                        Thread.sleep(20);
+                    } catch (IOException | InterruptedException e) {
+                        throw new RuntimeException(e);
+                    }
+                }
+            }
+        };
+        CacheLoader<Path, FSDataOutputStream> cacheLoader = new CacheLoader<Path, FSDataOutputStream>() {
+            @SuppressWarnings("NullableProblems")
+            @Override
+            public FSDataOutputStream load(Path path) throws Exception {
+                totalLoaderCount++;
+                try {
+                    return fileSystem.append(path);
+                } catch (IOException e){
+                    throw new IOException("File must be exist: " + path + "\n" + e);
+                }
+            }
+        };
+        cache = CacheBuilder.newBuilder()
+                .maximumSize(this.maxCapacity)
+                .concurrencyLevel(this.concurrentLevel)
+                .expireAfterAccess(this.maxStreamIdleMs, TimeUnit.MILLISECONDS)
+                .removalListener(listener)
+                .build(cacheLoader);
+    }
+
+    public long getSize() {
+        return cache.size();
+    }
+
+    public synchronized void dropEntry(Path path) {
```

Comment:
There is no need to synchronize on this method, especially since the close might take a while. Just invalidating the key should be enough.